### PR TITLE
Oyst tag conversion (#334)

### PR DESCRIPTION
**Pull request by rus-daniel**

### CODE REVIEW

**Comment by mehdichaouch** [comment]
@rus-daniel https://travis-ci.org/oystparis/oyst-1click-magento/jobs/368549051#L979 ;)

**Comment by mehdichaouch** [comment]
and https://travis-ci.org/oystparis/oyst-1click-magento/jobs/368549051#L1158 (@rus-daniel )

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 1]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+
+        if ($order->getOystOrderId()) {
+            $url .= 'oyst_oneclick/notifications/index';
+        } else {
+            $url .= 'checkout/onepage/success/';
+        }
+
+        $cookie = str_replace("\n", "", $this->execute('https://api.oyst.com/session'));
+
+        $json_array = [
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: you need to use ```array()``` to be valid with PHP 5.3
and
use camel case variable.

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 2]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+            "referer" => $url,
+            "tag" => "merchantconfirmationpage:display",
+            "oyst_cookie" => $cookie,
+            "user_agent" => $_SERVER['HTTP_USER_AGENT'],
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: Travis Error: 154 | ERROR   | Direct use of $_SERVER Superglobal detected.

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 3]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+        $order = $observer->getOrder();
+        $url = Mage::app()->getStore($order->getStoreId())->getBaseUrl(Mage_Core_Model_Store::URL_TYPE_LINK);
+
+        if ($order->getOystOrderId()) {
+            $url .= 'oyst_oneclick/notifications/index';
+        } else {
+            $url .= 'checkout/onepage/success/';
+        }
+
+        $cookie = str_replace("\n", "", $this->execute('https://api.oyst.com/session'));
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: use single quote

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 4]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+
+        $json_array = [
+            "referer" => $url,
+            "tag" => "merchantconfirmationpage:display",
+            "oyst_cookie" => $cookie,
+            "user_agent" => $_SERVER['HTTP_USER_AGENT'],
+            "cart_amount" => $order->getGrandTotal(),
+            "timestamp" => strtotime($order->getCreatedAt())
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: add comma to the last item (it's a good practice)

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 5]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+    {
+        $client = new Zend_Http_Client($url);
+
+        if ($data) {
+            $data = Zend_Json::encode($data);
+            $client->setRawData($data, 'application/json');
+        }
+        $response = $client->request(Zend_Http_Client::POST);
+
+        if($response->isError()) {
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: ```if (```

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 6]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+            Mage::log('Error: "' . $response->getMessage());
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: Use the oneclick log helper.

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 7]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+        if ($order->getOystOrderId()) {
+            $url .= 'oyst_oneclick/notifications/index';
+        } else {
+            $url .= 'checkout/onepage/success/';
+        }
+
+        $cookie = str_replace('\n', '', $this->execute('https://api.oyst.com/session'));
+
+        $json_array = array(
+            "referer" => $url,
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: change for single quote and use double quote only when you have to process what is inside the double quote

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 8]
```diff
@@ -128,4 +128,59 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+        );
+
+        $this->execute('https://api.staging.oyst.eu/events/oneclick', $json_array);
+    }
+
+    /**
+     * Execute http request.
+     *
+     * @param $url
+     * @param null $data
```
[the remark this answers is not in the document]
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Model/Observer.php]: ```@param null|mixed $data```

File: app/code/community/Oyst/OneClick/Model/Observer.php [hunk 9]
```diff
@@ -128,4 +128,60 @@ public function appendOneClickButton(Varien_Event_Observer $observer)
...
+     *
+     * @return null|mixed
```
Comment (mehdichaouch) [inline app/code/community/Oyst/OneClick/Model/Observer.php]: ```     * @return null|mixed```

**Comment by StefBrito** [comment]
Plus besoin géré par la data

**Pull request closed without merging**